hknweb/candidate/views.py

```diff
@@ -163,15 +158,22 @@ def send_cand_confirm_email(form):
     requester_name = challenge.requester.get_full_name()
     form = ChallengeConfirmationForm(request.POST or None, instance=challenge)
     context = {
-        'challenge' : challenge,
-        'requester_name' : requester_name,
+        'challenge': challenge,
+        'requester_name': requester_name,
         'form': form,
     }
 
     if form.is_valid():
         form.instance.reviewed = True
         form.save()
-        send_cand_confirm_email(form)
+        # csec has already confirmed, and now officer confirms
+        if challenge.officer_confirmed is True and challenge.csec_confirmed is True:
+            send_cand_confirm_email(request, form.instance, True)
+        # csec has not already rejected, and now officer rejects
+        elif challenge.officer_confirmed is False and challenge.csec_confirmed is not False:
```

Comment:
Another good place to reuse the confirmed() and rejected() methods I suggested earlier.

Reply:
This is slightly different though, you're checking if it was previously confirmed, and if it's now confirmed